arkiverer/src/test/kotlin/no/nav/soknad/arkivering/soknadsarkiverer/utils/ContainerizedKafka.kt

```diff
@@ -1,48 +1,90 @@
 package no.nav.soknad.arkivering.soknadsarkiverer.utils
 
-import no.nav.soknad.arkivering.soknadsarkiverer.config.*
+import no.nav.security.token.support.client.spring.ClientConfigurationProperties
+import no.nav.soknad.arkivering.soknadsarkiverer.kafka.KafkaConfig
+import org.apache.commons.compress.harmony.pack200.PackingUtils.config
+import org.junit.jupiter.api.AfterAll
 import org.junit.jupiter.api.fail
-import org.springframework.context.annotation.Bean
-import org.springframework.context.annotation.Configuration
-import org.springframework.context.annotation.Primary
+import org.springframework.beans.factory.config.YamlPropertiesFactoryBean
+import org.springframework.boot.context.properties.EnableConfigurationProperties
+import org.springframework.boot.context.properties.bind.Binder
+import org.springframework.boot.context.properties.source.ConfigurationPropertySource
+import org.springframework.boot.context.properties.source.MapConfigurationPropertySource
+import org.springframework.core.io.ClassPathResource
+import org.springframework.test.context.DynamicPropertyRegistry
+import org.springframework.test.context.DynamicPropertySource
 import org.testcontainers.containers.KafkaContainer
 import org.testcontainers.utility.DockerImageName
-import java.io.Closeable
+import java.util.Properties
 
-@Configuration
-class ContainerizedKafka : Closeable {
-	private val kafkaContainer: KafkaContainer = KafkaContainer(DockerImageName.parse("confluentinc/cp-kafka"))
-		.withNetworkAliases("kafka-broker")
 
-	init {
-		kafkaContainer.start()
-		listOf(kafkaMainTopic, kafkaProcessingTopic, kafkaMessageTopic, kafkaMetricsTopic)
-			.forEach { createTopic(it) }
-	}
+open class ContainerizedKafka  {
+
+
+
+	companion object {
+		private val kafkaConfig : KafkaConfig
+
+		init {
+			val factoryBean = YamlPropertiesFactoryBean()
+			factoryBean.setResources(ClassPathResource("application.yml"))
+
+			val properties: Properties? = factoryBean.getObject()
+
+			val propertySource: ConfigurationPropertySource = MapConfigurationPropertySource(properties)
+			val binder = Binder(propertySource)
+
+			kafkaConfig = binder.bind("kafka", KafkaConfig::class.java).get()
+		}
 
-	override fun close() {
-		kafkaContainer.stop()
-	}
+
+		// @Container
+		 val kafkaContainer: KafkaContainer = KafkaContainer(DockerImageName.parse("confluentinc/cp-kafka"))
+			.withNetworkAliases("kafka-broker")
+
+		@JvmStatic
+		@DynamicPropertySource
+		fun properties(registry: DynamicPropertyRegistry) {
+			start()
+			registry.add(
+				"kafka.brokers", kafkaContainer::getBootstrapServers
+			)
+		}
+
+	//	@JvmStatic
+	//	@BeforeAll
+		private fun start() {
+			kafkaContainer.start()
+
+		listOf(  kafkaConfig.topics.mainTopic, kafkaConfig.topics.processingTopic, kafkaConfig.topics.messageTopic, kafkaConfig.topics.metricsTopic)
+				.forEach { createTopic(it) }
+		}
+
+		@JvmStatic
+		@AfterAll
+		fun close() {
```

Comment:
Are you sure this method is called? Have you observed the output in the logs? IntelliJ reports it as unused, but the annotations can throw it off sometimes.

Reply:
Yep, very much called. https://howtodoinjava.com/junit5/after-all-annotation-example/